modules/RTC/ScreenObtainer.js

```diff
@@ -167,11 +174,36 @@ const ScreenObtainer = {
                 logger.info('Chrome extension not supported until ver 34');
 
                 return null;
+            } else if (navigator.getDisplayMedia
+                        && !options.desktopSharingChromeDisabled) {
+                return (ops, onSuccess, onFailure) => {
+                    navigator.getDisplayMedia({ video: true })
+                        .then(stream => {
+                            let applyConstraintsPromise;
+
+                            if (stream
+                                && stream.getTracks()
+                                && stream.getTracks().length > 0) {
+                                applyConstraintsPromise
+                                    = stream.getTracks()[0]
+                                        .applyConstraints(getSSConstraints());
+                            } else {
+                                applyConstraintsPromise = Promise.resolve();
```

Comment:
so if we get nothing we just want to resolve cleanly?  i guess that must be what the other flows do?

Reply:
Well, current chrome checks for stream === '' which indicates a cancel came from chrome extension, otherwise, fail.
Where Firefox just resolves with whatever is received and the same is for Edge getDisplayMedia. So not sure what to do?
I've added the check just to be safe when doing stream.getTracks()[0].